data/src/main/java/org/apache/iceberg/data/TableMigrationUtil.java

```diff
@@ -139,9 +213,16 @@ public static List<DataFile> listPartition(
                   buildDataFile(fileStatus.get(index), partitionKey, spec, metrics, "orc");
             });
       } else {
-        throw new UnsupportedOperationException("Unknown partition format: " + format);
+        if (skipOnError) {
+          LOG.warn("Skipping unknown partition format: {} - {}", format, partitionUri);
+        } else {
+          throw new UnsupportedOperationException(
+              String.format(
+                  "Unknown partition format: %s - %s, set skip_on_error=true to skip it.",
```

Comment:
This isn't the correct place to instruct the caller about how to avoid the error. This can be used through different paths, but only the procedure has `skip_on_error`.

Reply:
removed